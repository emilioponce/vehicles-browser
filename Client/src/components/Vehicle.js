import React from 'react'
import PropTypes from 'prop-types'
import TaxiInfo from './TaxiInfo'
import UberInfo from './UberInfo'
import styles from './Vehicle.module.css'
import { TAXI_TYPE, UBER_TYPE } from '../config/constants'
import uberImage from '../images/uber.svg'
import taxiImage from '../images/taxi.svg'

const Vehicle = props => {
  let { type } = props

  const getCarInfoComponent = type => {
    return type === TAXI_TYPE ? (
      <TaxiInfo title={type} position={props.coordinate} state={props.state} />
    ) : type === UBER_TYPE ? (
      <UberInfo
        title={type}
        position={props.coordinates}
        address={props.address}
        fuel={props.fuel}
        engineType={props.engineType}
        condition={
          'Exterior: ' + props.exterior + ' |Â Interior:' + props.interior
        }
        name={props.name}
      />
    ) : null
  }

  // conditional rendering depending on type
  let uberComponent = getCarInfoComponent(type)

  return (
    <div className={styles.vehicle}>
      <div className={styles.thumb}>
        <img
          src={
            type === TAXI_TYPE
              ? taxiImage
              : type === UBER_TYPE
              ? uberImage
              : null
          }
          className={styles.image}
          alt="vehicle"
        />
      </div>
      <div className={styles.info}>{uberComponent}</div>
    </div>
  )
}

Vehicle.propTypes = {
  type: PropTypes.string.isRequired,
  coordinate: PropTypes.object,
  state: PropTypes.string,
  address: PropTypes.string,
  fuel: PropTypes.number,
  engineType: PropTypes.string,
  condition: PropTypes.string,
  name: PropTypes.string
}

export default Vehicle
