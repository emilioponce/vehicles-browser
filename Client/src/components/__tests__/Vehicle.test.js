import React from 'react'
import ReactDOM from 'react-dom'
import Vehicle from '../Vehicle'
import renderer from 'react-test-renderer'

const component = (
  <Vehicle
    type="TAXI"
    coordinate={{ latitude: 41.5574, longitude: 2.0028 }}
    state="Exterior: GOOD | Interior:GOOD"
  />
)
describe('Component Vehicle', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('snapshot renders correctly', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
