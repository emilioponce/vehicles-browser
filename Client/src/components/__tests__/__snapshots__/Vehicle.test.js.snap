// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Vehicle snapshot renders correctly 1`] = `
<div
  className="vehicle"
>
  <div
    className="thumb"
  >
    <img
      alt="vehicle"
      className="image"
      src="taxi.svg"
    />
  </div>
  <div
    className="info"
  >
    <div
      className="taxiInfo"
    >
      <div
        className="title"
      >
        TAXI
      </div>
      <div>
        Position:
      </div>
      <div
        className="coordinates"
      >
        <div>
          Latitude: 
          41.5574
        </div>
        <div>
          Longitude: 
          2.0028
        </div>
      </div>
      <span
        className="inactiveState"
      >
        Exterior: GOOD | Interior:GOOD
      </span>
    </div>
  </div>
</div>
`;
